.github/workflows/build-dev-release.yml

```diff
@@ -115,6 +113,7 @@ jobs:
           ORG_GRADLE_PROJECT_signingKey: ${{ secrets.SONATYPE_GPG_ARMORED_KEY }}
           ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.SONATYPE_SIGNING_PASSWORD }}
   owasp-scan:
+    if: needs.build-and-publish-core-java.result == 'success' && !cancelled()
     needs:
       - build-and-publish-core-java
```

Comment:
note: I think we don't need the dependency to core build. OWASP scanning can be done in parallel. So, you can just use `!cancelled()` in my opinion. The same for another file